pkg/apis/hive/v1/hiveconfig_types.go

```diff
@@ -194,6 +214,67 @@ type ManageDNSAzureConfig struct {
 	ResourceGroupName string `json:"resourceGroupName"`
 }
 
+// ControllerGoroutinesConfig contains number of goroutines for each controller
+type ControllerGoroutinesConfig struct {
```

Comment:
Why are there different types for each config (e.g., `ControllerGoroutinesConfig` and `ControllerClientQPSConfig`) but not different types for each controller config (i.e., only one `ControllerConfig`)? Either everything can use the same type, or everything should use a different type.

Would it be any easier for a code standpoint and from a UX standpoint if these controller configurations were structured like the following?
```
type ControllerConfig struct {
  Gorotines *int32
  ClientQPS *int32
  ClientBurst *int32
  QueueQPS *int32
  QueueBurst *int32
}

type SpecificControllerConfig struct {
  Name string
  ControllerConfig
}

type ControllersConfig struct {
  Default *ControllerConfig
  Controllers []ControllerConfig
}

type HiveConfigSpec struct {
  ...
  ControllersConfig *ControllerConfig
}
```

Reply:
This looks way better than what I had implemented. Thanx!